rossum_api/api_client.py

```diff
@@ -319,7 +337,22 @@ async def _request(self, method: str, url: str, *args, **kwargs) -> httpx.Respon
             url = f"{self.base_url}/{url}"
         headers = kwargs.pop("headers", {})
         headers["Authorization"] = f"token {self.token}"
-        response = await self.client.request(method, url, headers=headers, *args, **kwargs)
+
+        for attempt, backoff in zip(
+            range(1, self.n_retries + 1), exponential_backoff(self.retry_backoff_factor)
+        ):
+            remaining_attempts = self.n_retries - attempt
+
+            try:
+                response = await self.client.request(method, url, headers=headers, *args, **kwargs)
+                if response.status_code in RETRIED_HTTP_CODES:
+                    continue
+                break
+            except httpx.RequestError:
+                if not remaining_attempts:
+                    raise
+            time.sleep(backoff)
```

Comment:
This is acync code, you should use `asyncio.sleep` in order not to block the execution loop.

Reply:
😱 thanks, I'm asyncio noob